crates/trie/sparse-parallel/src/trie.rs

```diff
@@ -678,13 +678,23 @@ impl ParallelSparseTrie {
         self.prefix_set = unchanged_prefix_set;
 
         // Update subtrie hashes in parallel
-        // TODO: call `update_hashes` on each subtrie in parallel
         let (tx, rx) = mpsc::channel();
-        for ChangedSubtrie { index, mut subtrie, mut prefix_set } in subtries {
-            subtrie.update_hashes(&mut prefix_set);
-            tx.send((index, subtrie)).unwrap();
-        }
+
+        let handles: Vec<_> = subtries
+            .into_iter()
+            .map(|ChangedSubtrie { index, mut subtrie, mut prefix_set }| {
+                let tx = tx.clone();
+                std::thread::spawn(move || {
+                    subtrie.update_hashes(&mut prefix_set);
+                    tx.send((index, subtrie)).unwrap();
+                })
+            })
+            .collect();
+
         drop(tx);
+        for handle in handles {
+            handle.join().unwrap();
+        }
```

Comment:
I think we need to think about how to spawn these, either rayon or reuse a blocking pool, so this likely requires more consideration